src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseGradingService.java

```diff
@@ -551,7 +547,6 @@ public void logReEvaluate(User user, ProgrammingExercise exercise, Course course
 
     /**
      * Filter all test cases from the score calculation that are never visible or ones with visibility "after due date" if the due date has not yet passed.
-     * @param testCases which should be filtered.
      * @return testCases, but the ones based on the described visibility criterion removed.
```

Comment:
Please mention if this method is in-place or not, i.e if I need to use the retunred Set or if the passed set will also be updated.

Reply:
I think since it returns a set it is clear that it is not in-place.
Also note the `@return` comment